src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -479,11 +479,14 @@ public bool IsAnalyzerSuppressed(DiagnosticAnalyzer analyzer)
 
                 if (hasDependentCompilationEndAction && forceAnalyzeAllDocuments != null)
```

Comment:
Yes, I can revert the change I did for perf-regression, that should fix this. I'll make the change after your changes go in. Thanks John!